towerlib/towerlib.py

```diff
@@ -1856,7 +1850,7 @@ def create_job_template(self,  # pylint: disable=too-many-arguments, too-many-lo
         if instance_groups:
             if not isinstance(instance_groups, (list, tuple)):
                 instance_groups = [instance_groups]
-            tower_instance_groups = [group_ for group_ in self.instance_groups]
+            tower_instance_groups = self.instance_groups
```

Comment:
Same goes here as the above comment. The groups are a generator not a list that is why we are doing the list comprehension. Make sure that it is not needed anywhere.

Reply:
Added the logic back to raise the exception